### PR TITLE
pkgs/build-support: stdenv.lib -> lib

### PR DESCRIPTION
**Pull request by siraben**
###### Motivation for this change
Part of #108938

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by Profpatsch — APPROVED** [review]
Looks reasonable, haven’t tried evaluating.

File: pkgs/build-support/release/ant-build.nix [hunk 1]
```diff
@@ -1,6 +1,6 @@
 { src
 , pkgs
-, stdenv ? pkgs.stdenv
+, lib, stdenv ? pkgs.stdenv
```
Comment (cole-h) [inline pkgs/build-support/release/ant-build.nix:3]:
```suggestion
, lib
, stdenv ? pkgs.stdenv
```

File: pkgs/build-support/cc-wrapper/default.nix [hunk 2]
```diff
@@ -6,7 +6,7 @@
 # compiler and the linker just "work".
 
 { name ? ""
-, stdenvNoCC
+, lib, stdenvNoCC
```
Comment (cole-h) [inline pkgs/build-support/cc-wrapper/default.nix:9]:
```suggestion
, lib
, stdenvNoCC
```

File: pkgs/build-support/bintools-wrapper/default.nix [hunk 3]
```diff
@@ -6,7 +6,7 @@
 # compiler and the linker just "work".
 
 { name ? ""
-, stdenvNoCC
+, lib, stdenvNoCC
```
Comment (cole-h) [inline pkgs/build-support/bintools-wrapper/default.nix:9]:
```suggestion
, lib
, stdenvNoCC
```

**Review by cole-h — COMMENTED** [review]
Nits:

File: pkgs/stdenv/darwin/default.nix [hunk 4]
```diff
@@ -81,6 +81,7 @@ in rec {
       mkCC = overrides: import ../../build-support/cc-wrapper (
         let args = {
           inherit shell;
+          inherit lib;
```
Comment (cole-h) [inline pkgs/stdenv/darwin/default.nix:84]:
```suggestion
          inherit lib shell;
```

**Review by cole-h — APPROVED** [review]
Diff LGTM, ofborg is green and a local `hydra-eval-jobs` is green. Thanks!

**Pull request merged**